src/devices/Media/SoundDevice/Devices/UnixSoundDevice.cs

```diff
@@ -0,0 +1,551 @@
+
+            Play(fs);
+        }
+
+        /// <summary>
+        /// Play WAV file.
+        /// </summary>
+        /// <param name="wavStream">WAV stream.</param>
+        public override void Play(Stream wavStream)
+        {
+            IntPtr @params = new IntPtr();
+            int dir = 0;
+            WavHeader header = GetWavHeader(wavStream);
+
+            OpenPlaybackPcm();
+            PcmInitialize(_playbackPcm, header, ref @params, ref dir);
+            WriteStream(wavStream, header, ref @params, ref dir);
+            ClosePlaybackPcm();
+        }
+
+        /// <summary>
+        /// Sound recording.
+        /// </summary>
+        /// <param name="second">Recording duration(In seconds).</param>
+        /// <param name="savePath">Recording save path.</param>
+        public override void Record(uint second, string savePath)
+        {
+            using FileStream fs = File.Open(savePath, FileMode.Create);
+
+            Record(second, fs);
+        }
+
+        /// <summary>
+        /// Sound recording.
+        /// </summary>
+        /// <param name="second">Recording duration(In seconds).</param>
+        /// <param name="saveStream">Recording save stream.</param>
+        public override void Record(uint second, Stream saveStream)
+        {
+            IntPtr @params = new IntPtr();
+            int dir = 0;
+            WavHeader header = new WavHeader
+            {
+                ChunkId = new[] { 'R', 'I', 'F', 'F' },
+                ChunkSize = second * Settings.RecordingSampleRate * Settings.RecordingBitsPerSample * Settings.RecordingChannels / 8 + 36,
+                Format = new[] { 'W', 'A', 'V', 'E' },
+                Subchunk1ID = new[] { 'f', 'm', 't', ' ' },
+                Subchunk1Size = 16,
+                AudioFormat = 1,
+                NumChannels = Settings.RecordingChannels,
+                SampleRate = Settings.RecordingSampleRate,
+                ByteRate = Settings.RecordingSampleRate * Settings.RecordingBitsPerSample * Settings.RecordingChannels / 8,
+                BlockAlign = (ushort)(Settings.RecordingBitsPerSample * Settings.RecordingChannels / 8),
+                BitsPerSample = Settings.RecordingBitsPerSample,
+                Subchunk2Id = new[] { 'd', 'a', 't', 'a' },
+                Subchunk2Size = second * Settings.RecordingSampleRate * Settings.RecordingBitsPerSample * Settings.RecordingChannels / 8
+            };
+
+            SetWavHeader(saveStream, header);
+
+            OpenRecordingPcm();
+            PcmInitialize(_recordingPcm, header, ref @params, ref dir);
+            ReadStream(saveStream, header, ref @params, ref dir);
+            CloseRecordingPcm();
+        }
+
+        private void SetWavHeader(Stream wavStream, WavHeader header)
+        {
+            Span<byte> writeBuffer2 = stackalloc byte[2];
+            Span<byte> writeBuffer4 = stackalloc byte[4];
+
+            wavStream.Position = 0;
```

Comment:
I'd avoid changing position in case we ever want to change this to stream it over the network or something that doesn't support seeking

Reply:
How should I modify it? If the method is executed according to the current process, there is no need to change the position of the stream.

Reply:
I think you should just remove this line and also make sure that after you're done reading you've read the whole header (position is exactly after the header but without using position explicitly)